components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -219,15 +222,18 @@ private IWAAuthenticatedUserBean userInformationInListedUserStores(String authen
                 IWAServiceDataHolder.getInstance().getMultiAttributeLoginService();
 
         if (multiAttributeLoginService.isEnabled(tenantDomain)) {
+            List<String> userStoreDomainsList = Arrays.stream(userStoreDomains.toUpperCase().split(","))
+                    .map(String::trim).collect(Collectors.toList());
             ResolvedUserResult resolvedUserResult = multiAttributeLoginService.resolveUser(
                     authenticatedUserName, tenantDomain);
             if (resolvedUserResult != null &&
                     ResolvedUserResult.UserResolvedStatus.SUCCESS.equals(resolvedUserResult.getResolvedStatus()) &&
-                    userStoreDomains.toUpperCase().contains(resolvedUserResult.getUser().getUserStoreDomain())) {
+                    userStoreDomainsList.contains(resolvedUserResult.getUser().getUserStoreDomain())) {
```

Comment:
is `resolvedUserResult.getUser().getUserStoreDomain()` null safe?

Reply:
This will not be null if the user is resolved from multi-attribute login successfully. Since we check for `ResolvedUserResult.UserResolvedStatus.SUCCESS.equals(resolvedUserResult.getResolvedStatus())` this is safe.